js/jquery.validation.js

```diff
@@ -438,6 +438,20 @@
                 this.message['notify'].remove();
             }
         }
+
+	setValuesFromResponse: function(response) {
+	    // Set form values from the response
+	    var $form = $(this.element);
+	    if (typeof response !== "undefined" && 'record' in response ) {
+		$.each(response.record, function (name, object) {
+		    var $element = $form.find('[name=' + name + ']');
+		    if ( 'valid' in object && object.valid && 'value' in object ) {
```

Comment:
In your codepen can you add the following types of form elements :
 * text input
 * select dropdown
 * checkbox with value = true
 * checkbox with value = "value1"

Then do some testing to check that this functions correctly sets the value of the different types of input above.

I suspect that .val() might not update the checkboxes as expected but could be wrong.